### PR TITLE
Disable sort controls when fetching data

### LINKED ISSUE #464: Sort toggle unpredictable in portrait on iPad
Sorting works as expected in landscape, but doesn't continue when switching to portrait

https://www.dropbox.com/s/ulga69jlwnt3h9q/iOS14%20-%20Sort%20toggle%20in%20portrait%20-%20iPad.mov?dl=0

### PR DESCRIPTION
**Pull request by VegetarianZombie**
At the moment, when the user taps the sorting button (popularity, most recent), the app will make a request to the server. This request can be very slow which means the end-user can keep tapping the sort button. This results with the sort state being out of sync with the actual returned results.

To solve this issue, I've locked the buttons. This means that the user won't be able to change the sort order once they tapped the button. I've changed the font of the button to signify it is no longer tappable. 

IMG_0108
IMG_0107
IMG_0110
IMG_0109

### CODE REVIEW

File: Emitron/Emitron/UI/Library/LibraryView.swift
Comment (sammyd) [inline Emitron/Emitron/UI/Library/LibraryView.swift]: It's a bit disappointing that "disabled" doesn't automatically have some kind of style modification that demonstrates that it is disabled.

File: Emitron/Emitron/UI/Library/LibraryView.swift
Comment (sammyd) [inline Emitron/Emitron/UI/Library/LibraryView.swift]: Is `isSorting` necessary? Could you not replicate the same functionality with

```swift
if [.loading, .loadingAdditional].contains(libraryRepository.state) {
```

Then it might be the case that you only need the changes in this function, and not the state var, or the change with `body`, or indeed the `changeSort()` below?

If this is the case, it might be worth adding a new property on the `LibraryRepository` (or `ContentRepository`) to proxy this "network operation in progress" concept.

I might however be overlooking things, in which case I apologise (=

**Review by sammyd — COMMENTED** [review]
Some questions about the approach before I approve this PR.

File: Emitron/Emitron/UI/Library/LibraryView.swift
Comment (VegetarianZombie) [reply Emitron/Emitron/UI/Library/LibraryView.swift]: Hi @sammyd,  yes - you are correct. I didn't need to provide an isSorting property. I originally thought I needed it so I can force a UI update. I removed the property and added your code as well as some additional code for the disabled check. It works exactly as before. 👍

I didn't add a property to the repository because I'm thinking I don't think I'll need to change the repo state in such a way. I was thinking I would add it if I needed to do so in a different part of the codebase. Do you think that's a good idea or would you prefer I add it to the repository? 

Thanks!

File: Emitron/Emitron/UI/Library/LibraryView.swift
Comment (VegetarianZombie) [reply Emitron/Emitron/UI/Library/LibraryView.swift]: Yes - it was pretty frustrating to see that the disabled state looks exactly the same as enabled.

File: Emitron/Emitron/UI/Library/LibraryView.swift
Comment (sammyd) [inline Emitron/Emitron/UI/Library/LibraryView.swift]: Is this import still necessary?

File: Emitron/Emitron/UI/Library/LibraryView.swift
Comment (sammyd) [inline Emitron/Emitron/UI/Library/LibraryView.swift]: Should probably be consistent with line 108. So either this, or switch 108 to match this `||` statement, whichever you feel is clearer.

```suggestion
      }.disabled([.loading, .loadingAdditional].contains(libraryRepository.state))
```

**Review by sammyd — APPROVED** [review]
:+1: Couple of questions, but other than that it looks good. Nice work—thank you Brian.

File: Emitron/Emitron/UI/Library/LibraryView.swift
Comment (sammyd) [reply Emitron/Emitron/UI/Library/LibraryView.swift]: I don't think that it's necessary to add a property anywhere unless you want to. The only thing I was kinda alluding to is the repetition of the "check the state" code that appears twice. I think it's fine as it is :+1:

File: Emitron/Emitron/UI/Library/LibraryView.swift
Comment (VegetarianZombie) [reply Emitron/Emitron/UI/Library/LibraryView.swift]: Sounds good - I've updated the PR with your suggestions. Thanks!

**Review by 0xTim — APPROVED** [review]
LGTM if all your questions have been answered @sammyd ?

**Review by sammyd — APPROVED** [review]
Yep! Looks good to me—thanks @VegetarianZombie & @0xTim

**Pull request merged**
